### PR TITLE
Bluetooth: Audio: Rename broadcaster and make opaque type

### PR DESCRIPTION
**Pull request by Thalley**
Rename `broadcaster` to `broadcast_source` to be more
similar to broadcast sink.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

**Comment by Thalley** [comment]
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/38260

File: include/bluetooth/audio.h [hunk 1]
```diff
@@ -590,8 +593,8 @@ struct bt_audio_capability_ops {
 	/** @brief Scan terminated callback
 	 *
 	 *  Scan terminated callback is called whenever a scan started by
-	 *  bt_audio_broadcaster_scan_start is terminated before
-	 *  bt_audio_broadcaster_scan_stop.
+	 *  bt_audio_broadcast_scan_start is terminated before
```
[the remark this answers is not in the document]
Comment (Thalley) [reply include/bluetooth/audio.h:596]: I think you are correct; you either need to add `()` or `@ref` to get the links. Thanks for pointing that out, will update

File: subsys/bluetooth/host/audio/chan.c [hunk 2]
```diff
@@ -73,11 +73,11 @@ struct bt_audio_base_ad {
 
 static struct bt_iso_cig *cigs[CONNECTED_AUDIO_GROUP_COUNT];
 static struct bt_audio_chan *enabling[CONFIG_BT_ISO_MAX_CHAN];
-static struct bt_audio_broadcaster broadcasters[BROADCAST_SRC_CNT];
+static struct bt_audio_broadcast_source broadcast_source[BROADCAST_SRC_CNT];
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/host/audio/chan.c:76]: Agreed

**Review by asbjornsabo — APPROVED** [review]
All my comments have been addressed.

**Pull request merged**